src/Workspaces/Core/Portable/Shared/Extensions/TypeInfoExtensions.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Shared.Extensions
+{
+    internal static class TypeInfoExtensions
+    {
+        public static ITypeSymbol GetConvertedTypeWithFlowNullability(this TypeInfo typeInfo)
+            => typeInfo.ConvertedType?.WithNullability(typeInfo.Nullability.FlowState);
+
+        public static ITypeSymbol GetConvertedTypeWithAnnotatedNullability(this TypeInfo typeInfo)
+            => typeInfo.ConvertedType?.WithNullability(typeInfo.Nullability.Annotation);
+
+        public static ITypeSymbol GetTypeWithFlowNullability(this TypeInfo typeInfo)
+            => typeInfo.Type?.WithNullability(typeInfo.Nullability.FlowState);
```

Comment:
I've already written this directly in a bunch of other places -- do you want to point those other ones to these extension methods?

Reply:
(fine to say "no" and do it as a follow up PR/commit)

Reply:
Done :) I left some areas that weren't as trivial to convert